app/src/main/java/com/veephealthoutloud/healthoutloud/Fragments/NewsfeedFragment.java

```diff
@@ -172,54 +164,19 @@ public void onActivityCreated(Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
     }
 
-    private ArrayList<IPost> GetPosts(){
-        // TODO: Change to use request to server when that's set up
-        ArrayList<IPost> list = new ArrayList<>();
-
-        ArrayList<String> feelings = new ArrayList<>();
-        feelings.add("sad");
-        feelings.add("frustrated");
-
-        getAllPosts();
-
-        IPost post1 = new Post("postID", "message", new Date(), feelings);
-        IPost post2 = new Post("postID2", "message2", new Date(), feelings);
-        IPost post3 = new Post("postID3", "message3", new Date(), feelings);
-        list.add(post1);
-        list.add(post2);
-        list.add(post3);
-        return list;
-    }
-
-    public void getAllPosts() {
-        final String postsURL = "http://healthoutloud-api.herokuapp.com/posts";
-
-        JsonArrayRequest allPosts = new JsonArrayRequest
-                (Request.Method.GET,
-                        postsURL,
-                        null,
-                        new Response.Listener<JSONArray>() {
-                            @Override
-                            public void onResponse(JSONArray response) {
-
-                                //ParsePosts(response);
-                                Log.d("Listener", "here");
-                                Log.d("Listener response", response.toString());
-                                Log.d("About to enter", "entered");
-                                ParsePosts(response);
-                            }
-                        }, new Response.ErrorListener() {
+    private void GetPosts(){
+        VolleyRequestsUtils.getAllPosts(getActivity().getApplicationContext(), new VolleyCallback() {
                     @Override
-                    public void onErrorResponse(VolleyError error) {
-                        Log.d("Error Handler", "here");
-                        // TODO: Error Handler
+                    public void onSuccess(JSONArray result) {
+                        ArrayList<IPost> allPosts = ParsePosts(result);
+                        postAdapter = new PostAdapter(getContext(), allPosts, R.menu.newsfeed_posts_popup);
+                        postListView.setAdapter(postAdapter);
```

Comment:
the post adapter should be created and set in the postListView in OnCreate, but here set the data source for the adapter with the list you get.